plugins/carla/CMakeLists.txt

```diff
@@ -0,0 +1,96 @@
+cmake_minimum_required(VERSION 3.16...3.25)
+
+option(ENABLE_CARLA "Enable building OBS with carla plugin host" ON)
+
+if(NOT ENABLE_CARLA)
+  if(OBS_CMAKE_VERSION VERSION_GREATER_EQUAL 3.0.0)
+    target_disable(carla)
+  else()
+    message(STATUS "OBS:  DISABLED   carla")
+  endif()
+  return()
+endif()
+
+# Find carla utils
+find_package(CarlaUtils)
+if(NOT CarlaUtils_FOUND)
+  if(OBS_CMAKE_VERSION VERSION_GREATER_EQUAL 3.0.0)
+    target_disable(carla)
+  else()
+    message(STATUS "OBS:  DISABLED   carla (carla-utils library not found)")
+  endif()
+  return()
+endif()
+
+# Find Qt
+find_qt(COMPONENTS Core Widgets)
+
+# Setup carla-bridge target
+add_library(carla-bridge MODULE)
+add_library(OBS::carla-bridge ALIAS carla-bridge)
+
+target_compile_definitions(carla-bridge PRIVATE CARLA_MODULE_ID="carla-bridge" CARLA_MODULE_NAME="Audio Plugin"
+                                                CARLA_UTILS_USE_QT)
+
+target_link_libraries(carla-bridge PRIVATE carla::utils OBS::libobs Qt::Core Qt::Widgets $<$<C_COMPILER_ID:GNU>:dl>)
+
+target_sources(
+  carla-bridge
+  PRIVATE carla.c
+          carla-bridge.cpp
+          carla-bridge-wrapper.cpp
+          common.c
+          pluginlistdialog.cpp
+          pluginrefreshdialog.hpp
+          qtutils.cpp)
+
+# non-mac/win systems should use rpath. this is typically automatically handled by pkg-config, but OBS does not read the
+# link flags from there
+if(NOT (OS_MACOS OR OS_WINDOWS))
+  set_property(
+    TARGET carla-bridge
+    APPEND
+    PROPERTY BUILD_RPATH "$<TARGET_FILE_DIR:carla::utils>")
+endif()
+
+if(OBS_CMAKE_VERSION VERSION_GREATER_EQUAL 3.0.0)
+  set_target_properties_obs(
+    carla-bridge
+    PROPERTIES AUTOMOC ON
+               AUTOUIC ON
+               AUTORCC ON
+               FOLDER plugins
+               PREFIX "")
+else()
+  set_target_properties(
+    carla-bridge
+    PROPERTIES AUTOMOC ON
+               AUTOUIC ON
+               AUTORCC ON
+               FOLDER plugins
+               PREFIX "")
+  setup_plugin_target(carla-bridge)
+endif()
+
+# Setup carla-patchbay target (only available for certain systems)
```

Comment:
What is `carla-patchbay` used for? It looks like this creates a single plugin for macOS and Windows, but two plugins for Linux. I'm not opposed to Linux having nice things, just trying to understand what it brings to the table.

I would also prefer if `carla-host-plugin` where a _component_ of `FindCarla.cmake` (so you could say `find_package(Carla REQUIRED utilities pluginhost)` and add a Linux-specific branch to the finder).

Then you'd use `Carla::pluginhost` to build the patchbay and don't need to bring package discovery code into the script here.

Reply:
> What is carla-patchbay used for? It looks like this creates a single plugin for macOS and Windows, but two plugins for Linux. I'm not opposed to Linux having nice things, just trying to understand what it brings to the table.

I explained elsewhere but carla-patchbay is a plugin for the full carla as a plugin, instead of being the "limited" approach of carla-bridge with just a plugin dialog.
A screenshot is easier to show: https://user-images.githubusercontent.com/1334853/235544415-6ae2c9e5-3cbf-47b2-8fc0-013f0ec8276d.png

This can be quite handy for loading multiple audio plugins in a single slot inside OBS, and have some complex routing between the audio plugins.

The main reason for being Linux only is due to it needing the full Carla installed somewhere that it can call upon. On Linux this is quite nice and easy, on macOS it becomes a problem due to code signing restrictions (I purposefully do not want Carla builds to be signed) and on Windows there is no installer at all for Carla, I always just ship a single zip.

> I would also prefer if carla-host-plugin where a component of FindCarla.cmake (so you could say find_package(Carla REQUIRED utilities pluginhost) and add a Linux-specific branch to the finder).

I would really, really, really rather not do that. pkg-config already handles everything needed, and this feature wont work for mac/win anytime soon. Adding an extra layer doesnt feel like a good use of our time.

Reply:
> This can be quite handy for loading multiple audio plugins in a single slot inside OBS, and have some complex routing between the audio plugins.

I leave this to others to decide whether we want this bonus complexity in the plugin. I am not aware of another filter that delegates actual functionality to an external program though it is akin to an physical external effects unit.

> I would really, really, really rather not do that. pkg-config already handles everything needed, and this feature wont work for mac/win anytime soon. Adding an extra layer doesnt feel like a good use of our time.

Fine by me, it's lean enough not to cause too much trouble. Maybe we rework this at a later point, but good enough for now.

Reply:
note that, as can be seen in the code, there is not much complexity to it. and the Carla GUI always runs on a separate process.
I have seen a **lot** of people using Carla together with OBS, but having to manage the 2 separately can sometimes be a pain.